lib/rules/no-unused-expressions.js

```diff
@@ -211,7 +161,8 @@ module.exports = {
 			ExpressionStatement(node) {
 				if (
 					Checker.isDisallowed(node.expression) &&
-					!isDirective(node)
+					!ignoreDirectives &&
+					!astUtils.isDirective(node)
 				) {
 					context.report({ node, messageId: "unusedExpression" });
 				}
```

Comment:
Same as https://github.com/eslint/eslint/pull/19644#discussion_r2053446917, this way `"ignoreDirectives": true` would ignore everything, not just directives:

```js
/* eslint no-unused-expressions: ["error", { "ignoreDirectives": true }] */

foo; // false negative

```

I think we'll need to keep the code that manually detects directives, as in the first version of this PR.